acceptance-tests/build.gradle

```diff
@@ -0,0 +1,155 @@
+import org.gradle.internal.os.OperatingSystem
+
+plugins {
+    id 'java'
+    id "de.undercouch.download" version "5.1.2"
+}
+
+ext {
+    athVersion = '5412.vc75c82cc0c93'
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(8))
+    }
+}
+
+group = 'org.jenkins-ci.plugins'
+description = 'Acceptance tests of Gradle plugin'
+
+repositories {
+    mavenCentral()
+    maven {
+        url 'https://repo.jenkins-ci.org/releases/'
+    }
+}
+
+configurations {
+    pluginsCollection {
+        canBeConsumed = false
+        canBeResolved = true
+    }
+}
+
+dependencies {
+    // same version as used by ATH
+    annotationProcessor "org.jenkins-ci:annotation-indexer:1.12"
+
+    implementation "org.jenkins-ci:acceptance-test-harness:${athVersion}"
+
+    testImplementation platform("io.netty:netty-bom:4.1.80.Final")
+    testImplementation "io.ratpack:ratpack-test:1.9.0"
+
+    pluginsCollection project(path: ':', configuration: 'pluginsCollectionDist')
+}
+
+def testDependenciesDir = new File("${buildDir}", "test-dependencies")
+
+def copyTestDependencies = tasks.create("copyTestDependencies", Copy) {
+    from configurations.pluginsCollection
+    into testDependenciesDir
+
+    include '*.hpi'
+    include '*.jpi'
+}
+
+def jenkinsVersions = [
+//    JenkinsVersion.LATEST, // Requires java 11 on CI
+    JenkinsVersion.LATEST_LTS,
+    JenkinsVersion.of('2.356')
+]
+
+def allTestTasks =
+    jenkinsVersions.collect { jenkinsVersion ->
+        def jenkinsWarLocation =
+            new File("${project.gradle.gradleUserHomeDir}/jenkins-cache/${jenkinsVersion.version}/jenkins.war")
+
+        def testTask =
+            jenkinsVersion.default
+                ? tasks.named("test")
+                : tasks.register("test_${jenkinsVersion.label}", Test)
+
+        testTask.configure {
+            dependsOn copyTestDependencies
```

Comment:
`dependsOn` is a bit of a smell. It's better to avoid it using proper inputs when possible.

`inputs.files(copyTestDependencies.outputs.files)`